oars_gb_pkg/helpers/path_planning/short_course_planner.py

```diff
@@ -0,0 +1,145 @@
+            vb_hyp = self.polar_effeciency(windangle, alpha)
+
+            # Projecting the max speed in the direction of the path
+            vt_test = np.dot(vb_hyp,path)
+
+            # Check if this velocity is the new max
+            if(vt_test>vt_max_R):
+                vt_max_R = vt_test
+                windangle_max_R = (windangle + alpha) % 360
+
+        # Analogous to the right side, but for the left side
+        for alpha in range (-180,0):
+            vb_hyp = self.polar_effeciency(windangle, alpha)
+            vt_test = np.dot(vb_hyp, path)
+            if (vt_test>vt_max_L):
+                vt_max_L = vt_test
+                windangle_max_L = (windangle + alpha) % 360
+
+        return vt_max_R,vt_max_L,windangle_max_R,windangle_max_L
+
+
+
+    def get_new_dir(self, vt_max_R, vt_max_L, windangle_max_R, windangle_max_L, p_c, path, boat_heading):
+        '''
+        checks whether right or left is better and returns the new direction (using the hysteresis)
+        :param vt_max_R: the maximum velocity on the right hand side
+        :param vt_max_L: the maximum velocity on the left hand side
+        :param windangle_max_R: angle corresponding to the max velocity on the right hand side
+        :param windangle_max_L: angle corresponding to the max velocity on the left hand side
+        :param p_c: beating parameter (controls the length of the tacks)
+        :param path: the path the boat is following
+        :param boat_heading: the boat's heading
+        :return: the new boat direction and speed
+        '''
+        # Calculates the hysteresis factor
+        n = 1 + (p_c / (abs(np.sqrt(path[0] ** 2 + path[1] ** 2))))
+
+        # Choose the direction that is closest to the current boat heading
+        R_diff = 180 - abs(abs(windangle_max_R - boat_heading) - 180);
+        L_diff = 180 - abs(abs(windangle_max_L - boat_heading) - 180);
+        if R_diff < L_diff:
+            # Choose the faster velocity
+            if vt_max_R * n < vt_max_L:
+                new_boat_heading = windangle_max_L
+            else:
+                new_boat_heading = windangle_max_R
+        else:
+            if vt_max_L * n < vt_max_R:
+                new_boat_heading = windangle_max_R
+            else:
+                new_boat_heading = windangle_max_L
+
+        return new_boat_heading
+
+    def run(self, Bp, Bh, T, w):
+        path = np.array(T) - np.array(Bp)
+        p_c = 20
+        # this runs above function to get the optimums and stores it into an array.
+        a,b,c,d = self.optimum(w, path)
+        new_dir = self.get_new_dir(a,b,c,d, p_c, path, Bh)
+
+        return new_dir
+
+if __name__ == '__main__':
+    windDir = int(input("wind angle: "))  # 90  #wind source direction
+    boatCurrDir = 0
+    targetList = [(0, 120), (120, 120), (0, 0), (120, 0), (120, -120), (0, -120), (0, 0)]
+    wind = Turtle()
```

Comment:
Maybe add a note as to what this is for (indicating the wind direction, right?).